### DISCUSSION TITLE
Immutable column value

### DISCUSSION THREAD

**Discussion by tmt-devq** [supabase/supabase#2878] [Questions] [answered]
Hi,

I've added an 'updated_at' column to a table and created a trigger to update to the current timestamp via the moddatetime extension whenever a new record gets inserted or an update is made. How can I prevent users from passing through a value when they post an update? I've tried it with a function and trigger like the below (just silently resetting the value but I will raise an exception here rather), but I am still able to insert a new record and specify an 'updated_at' value through the rest call?

```
create function tester() returns trigger language plpgsql as $$
begin
  new.updated_at = old.updated_at;
  return new;
end $$;

CREATE TRIGGER a_test_trigger BEFORE UPDATE ON items 
FOR EACH ROW EXECUTE PROCEDURE tester();
```

Thank you

**Comment by soedirgo** [comment, marked as answer]
If the `tester` function is `IMMUTABLE`, you can use [generated columns](https://www.postgresql.org/docs/current/ddl-generated-columns.html) instead of triggers:
```sql
ALTER TABLE items ADD updated_at timestamptz GENERATED ALWAYS AS (tester()) STORED;
```

Otherwise you can use an `AFTER` trigger on inserts and updates so that it overwrites the specified `updated_at`.
